rulesets/entityfilter/Predicates.cpp

```diff
@@ -125,7 +125,7 @@ bool ComparePredicate::isMatch(const QueryContext& context) const
                     const TypeNode* rightType =
                             static_cast<const TypeNode*>(right.Ptr());
                     if (rightType) {
-                        return rightType->isTypeOf(leftType);
+                        return leftType->isTypeOf(rightType);
```

Comment:
As soon as you've added some test for this I'll merge; I'm wary of doing it without there being test cases.